variables.tf

```diff
@@ -25,7 +25,13 @@ variable "enable_metrics" {
 
 variable "key_protect_instance_name" {
   type        = string
-  description = "The name of the existing Key Protect instance, if 'create_key_protect_instance' is true a new instance will be created with this name."
+  description = "The name of the Key Protect instance to provision."
+  default     = null
+}
+
+variable "existing_key_protect_instance_guid" {
+  type        = string
+  description = "The GUID of an existing Key Protect instance, required if 'var.create_key_protect_instance' is false."
```

Comment:
can we add variable validation in main.tf to ensure this is caught - aka if `key_protect_instance_name` is `null` then `existing_key_protect_instance_guid` cannot be `null` too